pkg/resource/pod_distruption_budget.go

```diff
@@ -0,0 +1,94 @@
+package resource
+
+import (
+	"errors"
+	"fmt"
+
+	policy "k8s.io/api/policy/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+// PdbBuilder models the PodDistruptionBudget that the
+// operator maintains.
+type PdbBuilder struct {
+	*Cluster
+
+	Selector map[string]string
+}
+
+// Build creates a policy.PodDisruptionBudget for the
+// StatefulSet.
+func (b PdbBuilder) Build(obj runtime.Object) error {
+	pdb, ok := obj.(*policy.PodDisruptionBudget)
+	if !ok {
+		return errors.New("failed to cast to PDB object")
+	}
+
+	// if we only have one Node we cannot have a PDB
+	// TODO we need to validate this in the CRD API
+	if b.Spec().Nodes == 1 {
+		return nil
+	}
+
+	// TODO fix or should we use this?
+	if pdb.ObjectMeta.Name == "" {
+		pdb.ObjectMeta.Name = b.DiscoveryServiceName()
+	}
+
+	if pdb.ObjectMeta.Labels == nil {
+		pdb.ObjectMeta.Labels = map[string]string{}
+	}
+
+	// TODO test sevice name
+	// Can we use the common label selector?
+	// commonLabels := labels.Common(cluster.Cr())
+	//  selector: commonLabels.Selector(),
+
+	labelSelector, err := metav1.ParseToLabelSelector("app=" + b.DiscoveryServiceName())
```

Comment:
Food for thought comment: Should we be looking move to `app.kubernetes.io/name` rather than `app`?

Reply:
I need to look at our labels.  We need to be looking for a label on the service that is super unique.

Reply:
I'm thinking I want to match the service name since that is guaranteed to be unique.

On the topic of labels.  We have this:

```yaml
  app.kubernetes.io/instance: crdb
```

We may want to change it to:

```yaml
  app.somecockroachurl/instance:crdb
```

Or add a label like the above.

Reply:
Probably something from `pkg/labels/label.go` could be helpful here.

Reply:
See https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/

Reply:
@vladdy yah I think that is what is being used, but for some reason, we are missing some of the labels.  What are your thoughts on adding a custom label?  Or should I just match the static labels?  For instance, the version label is not static.

Reply:
I really don't see why we need a custom label if k8s already recommends common ones. Also, if I remember it right this selector should also match the one used in the StatefulSet, so it would be better just to use the same way to specify the selector as it is done in StatefulSet reconciler.

Reply:
The updated code is pushed.  I am still doing some testing, but PTAL